src/py/flwr/server/driver/driver_servicer.py

```diff
@@ -56,12 +61,83 @@ def PushTaskIns(
         """Push a set of TaskIns."""
         log(INFO, "DriverServicer.PushTaskIns")
 
-        return PushTaskInsResponse(task_ids=[])
+        # Validate request
+        _raise_if(len(request.task_ins_set), "`task_ins_set` must not be empty")
+        for task_ins in request.task_ins_set:
+            _validate_incoming_task_ins(task_ins=task_ins)
+
+        # Store each TaskIns
+        task_ids: List[Optional[UUID]] = []
+        for task_ins in request.task_ins_set:
+            task_id: Optional[UUID] = self.driver_state.store_task_ins(
+                task_ins=task_ins
+            )
+            task_ids.append(task_id)
+
+        return PushTaskInsResponse(
+            task_ids=[str(task_id) if task_id else "" for task_id in task_ids]
+        )
 
     def PullTaskRes(
         self, request: PullTaskResRequest, context: grpc.ServicerContext
     ) -> PullTaskResResponse:
         """Pull a set of TaskRes."""
         log(INFO, "DriverServicer.PullTaskRes")
 
-        return PullTaskResResponse(task_res_set=[])
+        # Convert each task_id str to UUID
+        task_ids: Set[UUID] = {UUID(task_id) for task_id in request.task_ids}
+
+        # Read from state
+        task_res_set: List[TaskRes] = self.driver_state.get_task_res(
+            task_ids=task_ids, limit=10
```

Comment:
Why 10?

Reply:
Good catch - I removed the limit, it didn't make sense here